src/Voron/Impl/Paging/Pager.32.cs

```diff
@@ -99,12 +99,15 @@ private static void CleanupMemory(Pager pager, TxStateFor32Bits txState)
                     {
                         pager32BitsState.MemoryMapping.TryRemove(addr.StartPage, out set);
                     }
-                    
+
                     NativeMemory.UnregisterFileMapping(addr.File, addr.Address, addr.Size);
-                    var rc = Pal.rvn_unmap_memory(txState.Handle, (void*)addr.Address, addr.Size, out var errorCode);
-                    if (rc != PalFlags.FailCodes.Success)
+                    if (addr.State.Disposed is false) // we may closed the entire file during rollback
```

Comment:
> we may closed the entire file during rollback

Where does it happen? Can you please reference the relevant code?

Reply:
it is called from `RemoveInactiveScratches` during the rollback.
https://github.com/ravendb/ravendb/blob/42d4a96d08b492f3c50c2b409fbf898011bb6ff3/src/Voron/Impl/Scratch/ScratchBufferPool.cs#L363